src/puppetlabs/puppetdb/utils.clj

```diff
@@ -343,6 +360,29 @@
     {:command command :version (Integer. version)
      :certname certname :payload (json/->RawJsonString payload)}))
 
+(def ^Charset utf8 StandardCharsets/UTF_8)
+
+(defn utf8-bytes [s] (.getBytes ^String s utf8))
+
+(defn utf8-length
+  "Return the length in bytes of the given string when encoded in UTF-8"
+  [s]
+  (count (.getBytes ^String s utf8)))
+
+(defn utf8-truncate
+  "Truncate the given string such that its UTF-8 representation is at most
+  `max-bytes` long. Note that the returned string may be empty."
+  [s max-bytes]
+  (let [utf8-bytebuff (ByteBuffer/allocate max-bytes)
```

Comment:
The encoder is smart enough to stop if there's not enough room left in the buffer to encode a multi-byte character, as shown [in the tests](https://github.com/puppetlabs/puppetdb/pull/2065/files#diff-17ba5b2b53251cf7d606c36ba5254e2dR115).